--- gnome-control-center-2.28.1/panels/default-applications/gnome-default-applications.xml.in~	2009-05-03 21:30:56.000000000 +0300
+++ gnome-control-center-2.28.1/panels/default-applications/gnome-default-applications.xml.in	2010-03-09 22:29:32.991354516 +0200
@@ -147,6 +147,22 @@
       <run-in-terminal>false</run-in-terminal>
       <netscape-remote>false</netscape-remote>
     </web-browser>
+    <web-browser>
+        <_name>Google Chrome</_name>
+        <executable>google-chrome</executable>
+        <command>google-chrome %s</command>
+        <icon-name>google-chrome</icon-name>
+        <run-in-terminal>false</run-in-terminal>
+        <netscape-remote>false</netscape-remote>
+    </web-browser>
+    <web-browser>
+        <_name>Chromium Browser</_name>
+        <executable>chromium-browser</executable>
+        <command>chromium-browser %s</command>
+        <icon-name>chromium-browser</icon-name>
+        <run-in-terminal>false</run-in-terminal>
+        <netscape-remote>false</netscape-remote>
+    </web-browser>
   </web-browsers>
 
   <mail-readers>
