diff -aurN control-center-2.13.90.orig/capplets/default-applications/gnome-default-applications.xml.in control-center-2.13.90/capplets/default-applications/gnome-default-applications.xml.in
--- control-center-2.13.90.orig/capplets/default-applications/gnome-default-applications.xml.in	2006-01-28 19:52:24.000000000 +0100
+++ control-center-2.13.90/capplets/default-applications/gnome-default-applications.xml.in	2006-01-30 22:30:19.530754000 +0100
@@ -64,7 +64,7 @@
     <web-browser>
       <_name>Firefox</_name>
       <executable>firefox</executable>
-      <command>firefox %s</command>
+      <command>mozilla-firefox %s</command>
       <icon-name>firefox</icon-name>
       <run-in-terminal>false</run-in-terminal>
       <netscape-remote>true</netscape-remote>
@@ -234,6 +234,13 @@
       <icon-name>sylpheed</icon-name>
       <run-in-terminal>false</run-in-terminal>
     </mail-reader>
+    <mail-reader>
+      <_name>Sylpheed</_name>
+      <executable>sylpheed</executable>
+      <command>sylpheed --compose %s</command>
+      <icon-name>sylpheed</icon-name>
+      <run-in-terminal>false</run-in-terminal>
+    </mail-reader>
   </mail-readers>
 
   <terminals>
