diff -urN control-center-2.3.5.orig/configure.in control-center-2.3.5/configure.in
--- control-center-2.3.5.orig/configure.in	Thu Aug 14 23:00:28 2003
+++ control-center-2.3.5/configure.in	Thu Aug 14 23:17:06 2003
@@ -93,10 +93,10 @@
 AC_CHECK_LIB(Xrandr, XRRUpdateConfiguration,
   [AC_CHECK_HEADER(X11/extensions/Xrandr.h,
      have_randr=yes
-     RANDR_LIBS="-lXrandr -lXrender"
+     RANDR_LIBS="-lXrandr -lXrender -lXext"
      AC_DEFINE(HAVE_RANDR, 1, [Define if Xrandr extension is available]),
 	  :, [#include <X11/Xlib.h>])], : ,
-       -lXrandr -lXrender $x_libs)
+       -lXrandr -lXrender -lXext $x_libs)
 AM_CONDITIONAL(HAVE_RANDR, [test $have_randr = yes])
 	
 PKG_CHECK_MODULES(DISPLAY_CAPPLET, $COMMON_MODULES)
