diff -aurN control-center-2.13.90.orig/capplets/about-me/gnome-about-me.desktop.in.in control-center-2.13.90/capplets/about-me/gnome-about-me.desktop.in.in
--- control-center-2.13.90.orig/capplets/about-me/gnome-about-me.desktop.in.in	2006-01-23 12:36:20.000000000 +0100
+++ control-center-2.13.90/capplets/about-me/gnome-about-me.desktop.in.in	2006-01-30 22:34:18.650754000 +0100
@@ -7,7 +7,7 @@
 Terminal=false
 Type=Application
 StartupNotify=true
-Categories=GNOME;GTK;Settings;X-GNOME-PersonalSettings;
+Categories=GTK;GNOME;Settings;X-GNOME-PersonalSettings;X-Capplet;
 OnlyShowIn=GNOME;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=control-center
diff -aurN control-center-2.13.90.orig/capplets/appearance/data/gnome-appearance-properties.desktop.in.in control-center-2.13.90/capplets/appearance/data/gnome-appearance-properties.desktop.in.in
--- control-center-2.13.90.orig/capplets/appearance/data/gnome-appearance-properties.desktop.in.in	2006-01-23 12:36:20.000000000 +0100
+++ control-center-2.13.90/capplets/appearance/data/gnome-appearance-properties.desktop.in.in	2006-01-30 22:35:18.440754000 +0100
@@ -7,7 +7,7 @@
 Terminal=false
 Type=Application
 StartupNotify=true
-Categories=GNOME;GTK;Settings;DesktopSettings;
+Categories=GTK;GNOME;Settings;DesktopSettings;X-Capplet;
 OnlyShowIn=GNOME;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-control-center
diff -aurN control-center-2.13.90.orig/capplets/default-applications/default-applications.desktop.in.in control-center-2.13.90/capplets/default-applications/default-applications.desktop.in.in
--- control-center-2.13.90.orig/capplets/default-applications/default-applications.desktop.in.in	2006-01-23 12:36:20.000000000 +0100
+++ control-center-2.13.90/capplets/default-applications/default-applications.desktop.in.in	2006-01-30 22:35:43.670754000 +0100
@@ -7,7 +7,7 @@
 Terminal=false
 Type=Application
 StartupNotify=true
-Categories=GNOME;GTK;Settings;X-GNOME-PersonalSettings;
+Categories=GTK;GNOME;Settings;X-GNOME-PersonalSettings;X-Capplet;
 OnlyShowIn=GNOME;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=control-center
diff -aurN control-center-2.13.90.orig/capplets/display/display-properties.desktop.in.in control-center-2.13.90/capplets/display/display-properties.desktop.in.in
--- control-center-2.13.90.orig/capplets/display/display-properties.desktop.in.in	2006-01-23 12:36:20.000000000 +0100
+++ control-center-2.13.90/capplets/display/display-properties.desktop.in.in	2006-01-30 22:43:44.960754000 +0100
@@ -7,7 +7,7 @@
 Terminal=false
 Type=Application
 StartupNotify=true
-Categories=GNOME;GTK;Settings;HardwareSettings;
+Categories=GTK;GNOME;Settings;HardwareSettings;X-Capplet;
 OnlyShowIn=GNOME;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=control-center
diff -aurN control-center-2.13.90.orig/capplets/keybindings/keybinding.desktop.in.in control-center-2.13.90/capplets/keybindings/keybinding.desktop.in.in
--- control-center-2.13.90.orig/capplets/keybindings/keybinding.desktop.in.in	2006-01-23 12:36:20.000000000 +0100
+++ control-center-2.13.90/capplets/keybindings/keybinding.desktop.in.in	2006-01-30 22:45:11.700754000 +0100
@@ -7,7 +7,7 @@
 Terminal=false
 Type=Application
 StartupNotify=true
-Categories=GNOME;GTK;Settings;X-GNOME-PersonalSettings;
+Categories=GTK;GNOME;Settings;X-GNOME-PersonalSettings;X-Capplet;
 OnlyShowIn=GNOME;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=control-center
diff -aurN control-center-2.13.90.orig/capplets/keyboard/keyboard.desktop.in.in control-center-2.13.90/capplets/keyboard/keyboard.desktop.in.in
--- control-center-2.13.90.orig/capplets/keyboard/keyboard.desktop.in.in	2006-01-23 12:36:20.000000000 +0100
+++ control-center-2.13.90/capplets/keyboard/keyboard.desktop.in.in	2006-01-30 22:45:41.620754000 +0100
@@ -7,7 +7,7 @@
 Terminal=false
 Type=Application
 StartupNotify=true
-Categories=GNOME;GTK;Settings;HardwareSettings;
+Categories=GTK;GNOME;Settings;X-Capplet;
 OnlyShowIn=GNOME;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=control-center
diff -aurN control-center-2.13.90.orig/capplets/mouse/gnome-settings-mouse.desktop.in.in control-center-2.13.90/capplets/mouse/gnome-settings-mouse.desktop.in.in
--- control-center-2.13.90.orig/capplets/mouse/gnome-settings-mouse.desktop.in.in	2006-01-23 12:36:20.000000000 +0100
+++ control-center-2.13.90/capplets/mouse/gnome-settings-mouse.desktop.in.in	2006-01-30 22:45:55.270754000 +0100
@@ -7,7 +7,7 @@
 Terminal=false
 Type=Application
 StartupNotify=true
-Categories=GNOME;GTK;Settings;HardwareSettings;
+Categories=GTK;GNOME;Settings;X-Capplet;
 OnlyShowIn=GNOME;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=control-center
diff -aurN control-center-2.13.90.orig/capplets/network/gnome-network-preferences.desktop.in.in control-center-2.13.90/capplets/network/gnome-network-preferences.desktop.in.in
--- control-center-2.13.90.orig/capplets/network/gnome-network-preferences.desktop.in.in	2006-01-23 12:36:20.000000000 +0100
+++ control-center-2.13.90/capplets/network/gnome-network-preferences.desktop.in.in	2006-01-30 22:46:08.550754000 +0100
@@ -7,7 +7,7 @@
 Terminal=false
 Type=Application
 StartupNotify=true
-Categories=GNOME;GTK;Settings;X-GNOME-NetworkSettings;
+Categories=GTK;GNOME;Settings;X-GNOME-NetworkSettings;X-Capplet;
 OnlyShowIn=GNOME;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=control-center
diff -aurN control-center-2.13.90.orig/capplets/sound/gnome-settings-sound.desktop.in.in control-center-2.13.90/capplets/sound/gnome-settings-sound.desktop.in.in
--- control-center-2.13.90.orig/capplets/sound/gnome-settings-sound.desktop.in.in	2006-01-29 21:13:17.000000000 +0100
+++ control-center-2.13.90/capplets/sound/gnome-settings-sound.desktop.in.in	2006-01-30 22:46:22.590754000 +0100
@@ -7,7 +7,7 @@
 Terminal=false
 Type=Application
 StartupNotify=true
-Categories=GNOME;GTK;Settings;HardwareSettings;
+Categories=GTK;GNOME;Settings;X-Capplet;
 OnlyShowIn=GNOME;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=control-center
diff -aurN control-center-2.13.90.orig/capplets/windows/window-properties.desktop.in.in control-center-2.13.90/capplets/windows/window-properties.desktop.in.in
--- control-center-2.13.90.orig/capplets/windows/window-properties.desktop.in.in	2006-01-23 12:36:20.000000000 +0100
+++ control-center-2.13.90/capplets/windows/window-properties.desktop.in.in	2006-01-30 22:47:26.940754000 +0100
@@ -7,7 +7,7 @@
 Terminal=false
 Type=Application
 StartupNotify=true
-Categories=GNOME;GTK;Settings;DesktopSettings;
+Categories=GTK;GNOME;Settings;DesktopSettings;X-Capplet;
 OnlyShowIn=GNOME;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=control-center
