--- gnome-control-center-3.32.1/panels/user-accounts/cc-realm-manager.c	2012-09-24 17:39:34.000000000 +0200
+++ gnome-control-center-3.32.1.new/panels/user-accounts/cc-realm-manager.c	2012-10-05 14:30:29.557805038 +0200
@@ -23,7 +23,7 @@
 
 #include "um-realm-manager.h"
 
-#include <krb5/krb5.h>
+#include <krb5.h>
 
 #include <glib.h>
 #include <glib/gi18n.h>
@@ -703,8 +703,10 @@
         code = krb5_get_init_creds_opt_alloc (k5, &opts);
         g_return_val_if_fail (code == 0, code);
 
+#ifdef HAVE_KRB5_GET_INIT_CREDS_OPT_SET_OUT_CCACHE
         code = krb5_get_init_creds_opt_set_out_ccache (k5, opts, ccache);
         g_return_val_if_fail (code == 0, code);
+#endif
 
         code = krb5_get_init_creds_password (k5, &creds, principal,
                                              (char *)password,
