diff -aurN gnome-control-center-2.28.0.orig/configure.ac gnome-control-center-2.28.0/configure.in
--- gnome-control-center-2.28.0.orig/configure.ac	2009-09-24 11:34:15.000000000 +0200
+++ gnome-control-center-2.28.0/configure.ac	2009-09-24 11:38:03.000000000 +0200
@@ -298,6 +298,17 @@
                    enable_update_mimedb=yes)
 AM_CONDITIONAL(ENABLE_UPDATE_MIMEDB, test x$enable_update_mimedb = xyes)
 
+dnl =======================================
+dnl Update Desktop Database
+dnl =======================================
+
+AC_PATH_PROG(UPDATE_DESKTOP_DATABASE, update-desktop-database, no)
+
+AC_ARG_ENABLE(update-desktop,
+   AC_HELP_STRING([--disable-update-desktop],
+                   [do not update desktop database after installation]),,
+                   enable_update_desktop=yes)
+AM_CONDITIONAL(ENABLE_UPDATE_DESKTOP, test x$enable_update_desktop = xyes)
 
 AC_CONFIG_FILES([
 Makefile
diff -aurN gnome-control-center-2.28.0.orig/font-viewer/Makefile.am gnome-control-center-2.28.0/font-viewer/Makefile.am
--- gnome-control-center-2.28.0.orig/font-viewer/Makefile.am	2009-09-24 11:34:15.000000000 +0200
+++ gnome-control-center-2.28.0/font-viewer/Makefile.am	2009-09-24 11:41:45.000000000 +0200
@@ -28,11 +28,15 @@
 install-gconf-schemas:
 endif
 
+if ENABLE_UPDATE_DESKTOP
 install-desktop-database: install-desktopDATA
-	update-desktop-database $(DESTDIR)$(desktopdir)
+	$(UPDATE_DESKTOP_DATABASE) $(DESTDIR)$(desktopdir)
 
 uninstall-local:
 	rm -f $(DESTDIR)$(desktopdir)/mimeinfo.cache
+else
+install-desktop-database:
+endif
 
 @INTLTOOL_DESKTOP_RULE@
 @INTLTOOL_DIRECTORY_RULE@
diff -aurN gnome-control-center-2.28.0.orig/Makefile.am gnome-control-center-2.28.0/Makefile.am
--- gnome-control-center-2.28.0.orig/Makefile.am	2009-09-24 11:34:15.000000000 +0200
+++ gnome-control-center-2.28.0/Makefile.am	2009-09-24 11:36:11.000000000 +0200
@@ -46,7 +46,7 @@
 	MAINTAINERS \
 	gnome-doc-utils.make
 
-DISTCHECK_CONFIGURE_FLAGS = --disable-scrollkeeper --disable-update-mimedb
+DISTCHECK_CONFIGURE_FLAGS = --disable-scrollkeeper --disable-update-mimedb --disable-update-desktop
 
 GITIGNOREFILES=m4
 
