--- gnome-control-center-2.91.91/panels/common/Makefile.am.orig	2011-03-14 21:43:54.706666004 +0000
+++ gnome-control-center-2.91.91/panels/common/Makefile.am	2011-03-14 21:48:44.700000373 +0000
@@ -8,6 +8,7 @@
 	$(LIBLANGUAGE_CFLAGS)				\
 	-DDATADIR=\""$(datadir)"\"			\
 	-DUIDIR=\""$(pkgdatadir)/ui/user-accounts"\"	\
+	-DLIBDIR=\""$(libdir)"\"			\
 	-DLIBLOCALEDIR=\""$(prefix)/lib/locale"\"       \
 	-DGNOMELOCALEDIR=\""$(datadir)/locale"\"        \
 	-DUM_PIXMAP_DIR=\""$(pkgdatadir)/pixmaps"\"
--- gnome-control-center-2.91.91/panels/common/gdm-languages.c.orig	2011-03-14 21:38:11.669999713 +0000
+++ gnome-control-center-2.91.91/panels/common/gdm-languages.c	2011-03-14 21:47:55.906666990 +0000
@@ -47,7 +47,7 @@
 
 #define ALIASES_FILE DATADIR "/gdm/locale.alias"
 #define ARCHIVE_FILE LIBLOCALEDIR "/locale-archive"
-#define SYSTEM_ARCHIVE_FILE "/usr/lib/locale/locale-archive"
+#define SYSTEM_ARCHIVE_FILE LIBDIR "/locale/locale-archive"
 #define ISO_CODES_DATADIR ISO_CODES_PREFIX "/share/xml/iso-codes"
 #define ISO_CODES_LOCALESDIR ISO_CODES_PREFIX "/share/locale"
 
